Arrowgene.Ddon.GameServer/Handler/SkillGetAcquirableAbilityListHandler.cs

```diff
@@ -21715,9 +21715,15 @@ public override void Handle(GameClient client, StructurePacket<C2SSkillGetAcquir
             }
             else
             {
-                List<SecretAbility> UnlockedAbilities = _Database.SelectAllUnlockedSecretAbilities(client.Character.CommonId);
-                Response.AbilityParamList = AllSecretAbilities
-                    .Where(x => UnlockedAbilities.Contains((SecretAbility) x.AbilityNo)).ToList();
+                // If the CharacterId is 0, it is the current player, otherwise it is one of the players pawns
+                uint commonId = packet.Structure.CharacterId;
```

Comment:
Is this right? Common IDs dont necessarily match with Character/Pawn IDs

Reply:
So I tested with two different characters. When it was a player character, the ID would always be zero. If it was a pawn, it would be the character_common_id of the pawn.

Reply:
Are you sure it was the common id? I was sure that the common id was used only internally, and that it should've been pawn.PawnId instead

Reply:
Ah, you may be right, since my characters are in sequential order they probably match? So I should use the mechanism to query for them to get the proper pawn then, right?

Reply:
I was too fast and didn't see your reply. Should I do it this way instead of what I did?

Reply:
Nah, your code it does the same

Reply:
Maybe in future, I will add some function the the character manager which can return the `CharacterCommon` for us when we have these types of requests instead of typing this every time.